client/my-sites/earn/memberships/add-edit-plan-modal.jsx

```diff
@@ -261,202 +227,154 @@ const RecurringPaymentsPlanAddEditModal = ( {
 		closeDialog();
 	};
 
-	const renderGeneralTab = () => {
-		const editProduct = translate( 'Edit your existing payment plan.' );
-		const noProduct = translate(
-			'Each amount you add will create a separate payment plan. You can create many of them.'
-		);
-		return (
-			<>
-				<p>{ product && product.ID ? editProduct : noProduct }</p>
-				<FormFieldset>
-					<FormLabel htmlFor="currency">{ translate( 'Select price' ) }</FormLabel>
-					{ product && (
-						<Notice
-							text={ translate(
-								'Updating the price will not affect existing subscribers, who will pay what they were originally charged on renewal.'
-							) }
-							showDismiss={ false }
-						/>
-					) }
-					<FormCurrencyInput
-						name="currency"
-						id="currency"
-						value={ currentPrice }
-						onChange={ handlePriceChange }
-						currencySymbolPrefix={ currentCurrency }
-						onCurrencyChange={ handleCurrencyChange }
-						currencyList={ currencyList }
-						placeholder="0.00"
-					/>
-					{ ! isFormValid( 'price' ) && (
-						<FormInputValidation
-							isError
-							text={ translate( 'Please enter a price higher than %s', {
-								args: [
-									formatCurrency(
-										minimumCurrencyTransactionAmount(
-											currentCurrency,
-											connectedAccountDefaultCurrency
-										),
-										currentCurrency
-									),
-								],
-							} ) }
-						/>
-					) }
-				</FormFieldset>
-				<FormFieldset>
-					<FormLabel htmlFor="renewal_schedule">
-						{ translate( 'Select renewal frequency' ) }
-					</FormLabel>
-					<FormSelect id="renewal_schedule" value={ editedSchedule } onChange={ onSelectSchedule }>
-						<option value="1 month">{ translate( 'Renew monthly' ) }</option>
-						<option value="1 year">{ translate( 'Renew yearly' ) }</option>
-						<option value="one-time">{ translate( 'One time sale' ) }</option>
-					</FormSelect>
-				</FormFieldset>
+	const addPlan = editedPostsEmail
+		? translate( 'Create a paid newsletter subscription plan' )
+		: translate( 'Create a payment plan' );
+
+	const editPlan = editedPostsEmail
+		? translate( 'Edit a paid newsletter subscription plan' )
+		: translate( 'Edit a payment plan' );
+
+	const editing = product && product.ID;
+	return (
+		<Dialog
+			isVisible={ true }
+			onClose={ onClose }
+			buttons={ [
+				{
+					label: translate( 'Cancel' ),
+					action: 'cancel',
+				},
+				{
+					label: translate( 'Save' ),
+					action: 'submit',
+					disabled: ! isFormValid(),
+					isPrimary: true,
+				},
+			] }
+		>
+			<FormSectionHeading>{ product && product.ID ? editPlan : addPlan }</FormSectionHeading>
```

Comment:
You can shorten to `product?.ID` which will return falsy even if product isn't object or ID doesn't exist — as long as `product` itself isn't `undefined` which I don't think it should?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining